### PR TITLE
Feature/automated testing - closes #744

### LINKED ISSUE #744: Create one automated test case
Our code base currently has no automated unit or functional tests. If we can successfully create and execute one test case and run it against nightly deployment, we should be able to repeat it for remaining code base (or some portion of it).

Example test case: 
* Check whether search box appears on search result page. (Should not appear).
* Optionally, test whether is appears on front page. (Should appear.)

Choose the testing framework as well.

Definition of done
---
* [x] Testing framework is added to project.
* [x] Test exists for checking for search box.
  * [x] Test case exists for checking whether search box appears on search result page.
  * [x] Test case exists for checking whether search box appears on front page. (Currently testing`/catalogue` page. For front page need to use authorization module for "manual" authentication, which is not coded yet)
* [x] Test runs locally and can be demonstrated to the team.

### PR DESCRIPTION
**Pull request by frenchbread**

### CODE REVIEW

File: tests/jasmine/client/integration/sample/spec/hasSearchBox.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  beforeEach(function (done) {
+    Router.go('/catalogue');
+    Tracker.afterFlush(done);
+  });
+
+  beforeEach(waitForRouter);
+
+  it('should contain search form at "/catalogue" page', function () {
+
+    $(function () {
```
Comment (brylie) [inline tests/jasmine/client/integration/sample/spec/hasSearchBox.js]: Is it necessary to wrap these in sub functions? E.g. `$(function () {...});`

The [Meteor Velocity Jasmine testing examples](https://github.com/meteor-velocity/velocity-examples/blob/master/leaderboard-jasmine/tests/jasmine/client/integration/LeaderboardSpec.js#L23-L25) project does it slightly differently:

```js
 it("should show Grace above the give points button", function () {
    expect($("div.details > div.name").html()).toEqual("Grace Hopper");
  });
```

**Comment by brylie** [comment]
Make sure to pull in the recent changes from `develop`. There are concole errors about the `skin` helper when branding is empty.

**Comment by brylie** [comment]
Also, the assertion fails on my end. Did the assertion "should contain search form at "/catalogue" page" pass in your local branch?

File: tests/jasmine/client/integration/sample/spec/hasSearchBox.js (hunk 1 above)
Comment (frenchbread) [inline tests/jasmine/client/integration/sample/spec/hasSearchBox.js]: @brylie That's what we were talking about earlier. This is the same as `JQuery.ready()` - waits until all elements on a page are loaded.

**Comment by frenchbread** [comment]
Alter adding `$(function(){ ... });` all (two) tests passed successfully. Which one fails for you`?

**Comment by frenchbread** [comment]
Resolved conflicts.

The test case which is testing `/catalogue` page still fails at first. It only passes when the meteor server reloads itself automatically after changes made to the code. This happens even being wrapped to `$(document).ready(function(){ ... })`. 

@brylie Any ideas?

**Comment by brylie** [comment]
Please review my commits. The code is now simpler, and slightly easier to read. Both tests pass on my local development environment.

**Comment by frenchbread** [comment]
@brylie First test still does not pass on my local env.

screen shot 2016-01-25 at 15 04 55

**Comment by bajiat** [comment]
@ashakunt Do you have any suggestions on what might be the problem here? Please see the discussion above.

**Comment by brylie** [comment]
Hey @frenchbread check out [jasmine-jquery plugin](https://github.com/velesin/jasmine-jquery). It might make our DOM checks a bit more semantic, without requiring the Array.length checks, etc. The jQuery dependency is alright, since jQuery has a clean API and is a central part of our project.

**Comment by brylie** [comment]
Here is a video demonstrating Intern.js setup, testing, etc:

[Intern video](http://img.youtube.com/vi/_KFjuEKLqDA/maxresdefault.jpg)

**Comment by brylie** [comment]
The previous video recommends using a library called Leadfoot for simulating user interaction. Leadfoot has methods relating to [DOM Elements](https://theintern.github.io/leadfoot/Element.html), such as [findById](https://theintern.github.io/leadfoot/Command.html#findById). The DOM Element may be compatible with assertion libraries, e.g. `expect(element).toExist()`.

**Comment by brylie** [comment]
@frenchbread here is the fixed video link:
[Intern video](https://www.youtube.com/watch?v=_KFjuEKLqDA)

Any status update on this PR?

**Comment by frenchbread** [comment]
@brylie Working in [`feature/theintern-automated-testing`](https://github.com/apinf/api-umbrella-dashboard/tree/feature/theintern-automated-testing) branch.

**Comment by bajiat** [comment]
@brylie Can you check what to do with the issue and the PR?

File: .meteor/packages [hunk 2]
```diff
@@ -92,5 +92,7 @@ frenchbread:elastic-rest
 universe:reactive-queries
 ecmascript
 simple:reactive-method
+sanjo:jasmine
+velocity:html-reporter
```
Comment (brylie) [inline .meteor/packages]: Lets see if we can do unit/functional testing without Velocity.

File: tests/jasmine/client/integration/lib/wait_for_router_helper.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
+(function (Meteor, Tracker, Router) {
```
Comment (brylie) [inline tests/jasmine/client/integration/lib/wait_for_router_helper.js]: After the Meteor 1.3 upgrade, we can use ES2015 `import` to make sure these dependencies are present in the test.

**Comment by brylie** [comment]
Closing, since this task is not currently in progress. We can leave the branch open.

**Pull request closed without merging**